accelerator/runtime_cli.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef ACCELERATOR_RUNTIME_CLI_H
+#define ACCELERATOR_RUNTIME_CLI_H
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file accelerator/runtime_cli.h
+ * @brief Runtime command-line utility
+ */
+
+/**
+ * @brief Runtime command-line utility thread
+ */
+void *cli_routine(void *arg);
+
+/**
+ * @brief Command state types in state machine
+ *
+ * - CMD_LAST: The last subcommand, could be followed by zero or more parameters
+ * - CMD_STATE: Subcommands expected, should step to next state according to `next_state`
+ */
+enum { CMD_LAST, CMD_STATE };
+struct cli_cmd;
```

Comment:
Could we integrate the declaration and definition below together?

Reply:
I'm afraid not, those two structures depend on each other, so we need to use forward declaration.